geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureWriterTest.scala

```diff
@@ -115,19 +115,19 @@ class AvroSimpleFeatureWriterTest extends Specification with Mockito with Abstra
       // 1 key  and 0 values have type Integer
       there was one(encoder).writeString("java.lang.Boolean")
 
-      // 1 key  and 1 value  are null
-      there was two(encoder).writeString(AvroUserDataSerialization.NullMarkerString)
+      // 1 key  and 1 value  are null - results in 2 null calls for each, class name + value
+      there was 4.times(encoder).writeNull()
 
       // visibility data
       there was one(encoder).writeString(SecurityUtils.FEATURE_VISIBILITY)
       there was one(encoder).writeString(vis)
 
       // hint data
       there was one(encoder).writeString(HintKeySerialization.keyToId(Hints.USE_PROVIDED_FID))
-      there was one(encoder).writeBoolean(true)
+      there was one(encoder).writeString("true")
 
       // key = 5, value = null
-      there was one(encoder).writeInt(5)
+      there was one(encoder).writeString("5")
```

Comment:
Hmm... why did this change from Int to String?  (Just finishing my first skim through.)

Reply:
the avro user-data schema is defined as only being strings

Reply:
Ah.... and that may explain the pendingUntilFixed thing in the unit test...